### PR TITLE
TST: fix fixture for numpy dtypes

### PR DESCRIPTION
**Pull request by h-vetinari**
- [x] split off from #23982
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

This was one of the things I stumbled over while working on #23982. There's a fixture for all_numpy_dtypes that got introduced in #21899, but it's a lie - it doesn't cover (nearly) all numpy dtypes. It's currently only used in one test, and so I've adapted it here to be able to actually test all numpy dtypes.

### CODE REVIEW

File: pandas/tests/series/test_duplicates.py [hunk 1]
```diff
@@ -91,22 +91,17 @@ def __ne__(self, other):
...
-@pytest.mark.parametrize('keep, expected',
-                         [('first', Series([False, False, True, True])),
-                          ('last', Series([True, True, False, False])),
-                          (False, Series([True, True, True, True]))])
-def test_drop_duplicates_bool(keep, expected):
-    tc = Series([True, False, True, False])
+def test_drop_duplicates(any_numpy_dtype, keep, expected):
+    tc = Series([1, 0, 3, 5, 3, 0, 4], dtype=np.dtype(any_numpy_dtype))
+
+    if tc.dtype == 'bool':
```
Comment (jreback) [inline pandas/tests/series/test_duplicates.py]: i would just skip if bool here

File: pandas/conftest.py [hunk 2]
```diff
@@ -386,8 +386,16 @@ def tz_aware_fixture(request):
 COMPLEX_DTYPES = [complex, "complex64", "complex128"]
 STRING_DTYPES = [str, 'str', 'U']
 
+DATETIMEDELTA_DTYPES = ['datetime64[ns]', 'M8[ns]',
```
Comment (jreback) [inline pandas/conftest.py]: separate these out, these should not be grouped together

**Comment by jreback** [comment]
lgtm. ping on green.

**Comment by h-vetinari** [comment]
@jreback
ping on green

**Comment by jreback** [comment]
thanks

**Pull request merged**